src/Cake.Npm.Tests/AddUser/NpmAddUserTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using Xunit;
+using Cake.Npm.AddUser;
+
+namespace Cake.Npm.Tests.AddUser
+{
+    public class NpmAddUserTests
+    {
+        public sealed class TheAddUserMethod
+        {
+            [Fact]
+            public void Should_Redirect_Standard_Error()
+            {
+                var fixture = new NpmAddUserFixture();
+                fixture.Settings.RedirectStandardError = true;
+
+                var result = fixture.Run();
+
+                Assert.True(result.Process.RedirectStandardError);
+            }
+
+            [Fact]
+            public void Should_Throw_If_Settings_Are_Null()
+            {
+                // Given
+                var fixture = new NpmAddUserFixture();
+                fixture.Settings = null;
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                result.IsArgumentNullException("settings");
+            }
+            
+            [Fact]
+            public void Should_Add_Registry_Url_To_Arguments()
+            {
+                // Given
+                var fixture = new NpmAddUserFixture();
+                fixture.Settings.FromRegistry(new Uri("https://registry.com"));
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("adduser --registry https://registry.com/ --auth-type legacy", result.Args);
+            }
+
+            [Fact]
+            public void Should_Throw_If_Not_Absolute_Uri()
+            {
+                // Given
+                var fixture = new NpmAddUserFixture();
+
+                // When
+                var result = Record.Exception(
+                    () =>
+                        fixture.Settings.FromRegistry(new Uri("https://www.example.com/foo/mypackage.tgz").MakeRelativeUri(new Uri("https://www.example.com"))));
```

Comment:
This is a nice check (which we also should do for the other settings classes accepting a registry parameter). But there's currently no implementation which checks that Url is absolute, therefore the failing test